google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1beta2/st/STBigQueryStorageLongRunningWriteTest.java

```diff
@@ -0,0 +1,245 @@
+            ServiceOptions.getDefaultProjectId(), DATASET, TABLE);
+    tableId2 =
+        String.format(
+            "projects/%s/datasets/%s/tables/%s",
+            ServiceOptions.getDefaultProjectId(), DATASET, TABLE2);
+    LOG.info(
+        String.format(
+            "%s tests running with parent project: %s",
+            ITBigQueryStorageLongRunningTest.class.getSimpleName(), parentProjectId));
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    if (client != null) {
+      client.close();
+    }
+    if (bigquery != null) {
+      RemoteBigQueryHelper.forceDelete(bigquery, DATASET);
+      LOG.info("Deleted test dataset: " + DATASET);
+    }
+  }
+
+  @Test
+  public void testDefaultStream() throws IOException, InterruptedException, ExecutionException,
+      Descriptors.DescriptorValidationException {
+    // Set up a default stream. Write to it for a long time, (a few minutes for now) and make
+    // sure that everything goes well.
+    String tableName = "JsonTableDefaultStream";
+    TableInfo tableInfo =
+        TableInfo.newBuilder(
+            TableId.of(DATASET, tableName),
+            StandardTableDefinition.of(
+                Schema.of(
+                    com.google.cloud.bigquery.Field.newBuilder(
+                        "test_str", StandardSQLTypeName.STRING)
+                        .build(),
+                    com.google.cloud.bigquery.Field.newBuilder(
+                        "test_numerics", StandardSQLTypeName.NUMERIC)
+                        .setMode(Field.Mode.REPEATED)
+                        .build(),
+                    com.google.cloud.bigquery.Field.newBuilder(
+                        "test_datetime", StandardSQLTypeName.DATETIME)
+                        .build())))
+            .build();
+    bigquery.create(tableInfo);
+    TableName parent = TableName.of(ServiceOptions.getDefaultProjectId(), DATASET, tableName);
+    try (JsonStreamWriter jsonStreamWriter =
+        JsonStreamWriter.newBuilder(parent.toString(), tableInfo.getDefinition().getSchema())
+            .createDefaultStream()
+            .setBatchingSettings(
+                StreamWriter.Builder.DEFAULT_BATCHING_SETTINGS
+                    .toBuilder()
+                    .setRequestByteThreshold(1024 * 1024L) // 1 Mb
+                    .setElementCountThreshold(2L)
+                    .setDelayThreshold(Duration.ofSeconds(2))
+                    .build())
+            .build()) {
+      for (int i = 0; i < 5; i++){
+        LOG.info("Sending a message");
+        // Ramping up the size increases the latency
+        JSONObject row = MakeJsonObject(1);
+        JSONArray jsonArr = new JSONArray(new JSONObject[] {row});
+
+        LocalDateTime start = LocalDateTime.now();
+        Date startTime = new Date();
+        ApiFuture<AppendRowsResponse> response = jsonStreamWriter.append(jsonArr, -1);
+        assertEquals(0, response.get().getAppendResult().getOffset().getValue());
```

Comment:
This is all synchronized calls, which is not what we want. You can add a TODO about making it into asynchronized calls.

Reply:
I've added the TODO